runtime/gc_base/ObjectAccessBarrier.cpp

```diff
@@ -1546,7 +1574,22 @@ MM_ObjectAccessBarrier::copyObjectFields(J9VMThread *vmThread, J9Class *objectCl
 		while (offset < limit) {
 			UDATA srcAddress = (UDATA)srcObject + srcOffset + offset;
 			UDATA destAddress = (UDATA)destObject + destOffset + offset;
-			if (sizeof(uint32_t) == referenceSize) {
+			bool copy64Bits = false;
+			if (isValueType
+				&& (sizeof(uint32_t) == referenceSize)
+				&& (0 == (srcAddress & 7))
+				&& ((offset + referenceSize) < limit)
+			) {
+				copy64Bits = true;
+			}
+			if (copy64Bits) {
+				*(uint64_t *)destAddress = *(uint64_t *)srcAddress;
+				/**
+				 * When doing 64-bit copy, offset needs to be advanced 8 bytes. referenceSize is 4 bytes here.
+				 * Advance offset 4 bytes here. Another 4 bytes are advanced at the end of the while loop below.
+				 */
+				offset += referenceSize;
+			} else if (sizeof(uint32_t) == referenceSize) {
```

Comment:
Now that the code has grown, is there a way to share mixed and primitive only paths? Is there a known benefit of special treating primitive only fields case?

Reply:
I guess the reason why they were not shared is because of the extra descriptionBits lookup in the mixed mode. Not sure sharing them  in this function makes the code easier to read. Or you want to make the sharable part into a new function ? The only sharable piece might be: 

```
if (copy64Bits) {
	*(uint64_t *)destAddress = *(uint64_t *)srcAddress;
	offset += sizeof(uint64_t);
} else {
	*(uint32_t *)destAddress = *(uint32_t *)srcAddress;
	offset += sizeof(uint32_t);
}
```

Reply:
on a second look, you are right, there is not much to share (diffs were confusing me)